app/services/DepartureP5MessageService.scala

```diff
@@ -43,9 +43,9 @@ class DepartureP5MessageService @Inject() (
               messagesForMovement.messages.find(_.messageType == DepartureNotification) match {
                 case Some(ie015) =>
                   for {
-                    lrn <- departureMovementP5Connector.getLRN(ie015.bodyPath).map(_.referenceNumber)
-                    ie056  = messagesForMovement.messages.find(_.messageType == RejectedByOfficeOfDeparture)
-                    xPaths = ie056.map(_.functionalErrors.map(_.errorPointer))
+                    lrn   <- departureMovementP5Connector.getLRN(ie015.bodyPath).map(_.referenceNumber)
+                    ie056 <- getRejectionMessage(ie015.departureId)
+                    xPaths = ie056.map(_.data.functionalErrors.map(_.errorPointer))
                     isDeclarationAmendable <- xPaths.filter(_.nonEmpty).fold(Future.successful(false))(cacheConnector.isDeclarationAmendable(lrn, _))
                   } yield DepartureMovementAndMessage(movement, messagesForMovement, lrn, isDeclarationAmendable)
```

Comment:
You may be able to flatten this using OptionT like so:

        (
          for {
          messagesForMovement    <-  OptionT.liftF(departureMovementP5Connector.getMessagesForMovement(movement.messagesLocation))
          ie015                  <-  OptionT.fromOption[Future](messagesForMovement.messages.find(_.messageType == DepartureNotification))
          lrn                    <-  OptionT.liftF(departureMovementP5Connector.getLRN(ie015.bodyPath).map(_.referenceNumber))
          ie056                  <-  OptionT.liftF(getRejectionMessage(ie015.departureId))
          xPaths                 = ie056.map(_.data.functionalErrors.map(_.errorPointer))
          isDeclarationAmendable <- OptionT.liftF(xPaths.filter(_.nonEmpty).fold(Future.successful(false))(cacheConnector.isDeclarationAmendable(lrn, _)))
        } yield DepartureMovementAndMessage(movement, messagesForMovement, lrn, isDeclarationAmendable)
       ).getOrElse(Future.failed(new Throwable("Movement did not contain an IE015 message")))

Reply:
I considered this, however if any one of those futures fail we will get the "Movement did not contain an IE015 message" throwable